File: _h2d_test/examine_hourly_2_daily_sum.py

"""
This script just verifies that the daily rainfall time series retrieved from AQ should be
literally offset (backwards) by 1 day: i.e., say `ts` is the daily time series from AQ,
its real daily rainfall time series should be ts.index = ts.index - pd.Timedelta('1D')
"""


import datetime
from pathlib import Path

import polars as pl
import _tools.fun_s_pl as fpl



p = Path('.')
pp = p / '_h2d_test'
ppp = pp / 'data'


hwl = pl.read_parquet(ppp / 'hRain_wide.parquet')
hwl = hwl.with_columns(pl.col('Time').str.to_datetime('%Y-%m-%d %H:%M:%S'))

dwl = pl.read_parquet(ppp / 'dRain_wide.parquet')
dwl = dwl.with_columns(pl.col('Date').str.to_date('%Y-%m-%d'))



# After the fix, their time sequence should be matched
dwl_fixed = dwl.with_columns(pl.col('Date').sub(datetime.timedelta(days=1)).alias('Date'))
dwl_f = fpl.hourly_2_daily(hwl, agg=pl.sum).drop('Site')

File: _h2d_test/examine_hourly_2_daily_sum.R

rm(list = ls(all.names = TRUE))

source("_tools/fun_s.R")

fi <- "_h2d_test/data/hRain_wide.parquet"
h <- arrow::read_parquet(fi)

h$Time <- as.POSIXct(h$Time, tz = "Etc/GMT-12", format = "%Y-%m-%d %H:%M:%S")

d <- hourly_2_daily(h, agg = sum)

# Load the daily time series from AQ
d_AQ <- arrow::read_parquet("_h2d_test/data/dRain_wide.parquet")
d_AQ$Date <- as.Date(d_AQ$Date, format = "%Y-%m-%d")
